processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/LimitedTemporaryStorage.java

```diff
@@ -125,10 +125,10 @@ public void close()
       for (File file : ImmutableSet.copyOf(files)) {
         delete(file);
       }
-      files.clear();
-      if (storageDirectory.exists() && !storageDirectory.delete()) {
+      if (files.size() > 0 && storageDirectory.exists() && !storageDirectory.delete()) {
```

Comment:
`size` must actually check the object, and depending on the collection, may have to traverse or do other special record keeping operations. Some collections can optimize this and others cannot. `isEmpty()` should be used when simply checking for emptyness.

Reply:
Good catch. Thanks.